drape/glyph_manager.cpp

```diff
@@ -258,16 +258,22 @@ class Font
       // This height gives 16 instead of expected 22-23.
       // int32_t const height = metrics.height >> 6;
 
-      outMetrics.AddGlyphMetrics(static_cast<int16_t>(fontIndex), glyphId, xOffset, yOffset, xAdvance, GetLineHeigthPixels());
+      outMetrics.AddGlyphMetrics(static_cast<int16_t>(fontIndex), glyphId, xOffset, yOffset, xAdvance,
+                                 GetLineHeigthPixels(metrics.vertAdvance));
     }
   }
 
-  int32_t GetLineHeigthPixels() const
+private:
+  int32_t GetLineHeigthPixels(FT_Pos vertAdvance) const
   {
-    auto const vertAdvance = m_fontFace->glyph->metrics.vertAdvance;
-    auto const yMax = vertAdvance && vertAdvance < m_fontFace->bbox.yMax ? vertAdvance : m_fontFace->bbox.yMax;
-    int32_t const bboxYMax = FT_MulFix(yMax, m_fontFace->size->metrics.y_scale) >> 6;
+    // Freetype says that vertAdvance is unreliable if it contains a synthesized value in case of FT_HAS_VERTICAL
+    // returning false. Testing on supported fonts shows that non-zero vertAdvance can be trusted as a line height.
+    if (vertAdvance && vertAdvance < m_fontFace->bbox.yMax) [[likely]]
```

Comment:
Expicitly vertAdvance != 0 (or probably vertAdvance > 0, IDK)
[[likely]] :)

Reply:
Thanks, fixed. 

What's funny about [[likely]]? It's called hundreds of times per second.

Reply:
I don't think that [likely] can change something in this code and expect to have some reasonable tradeoff between these attributes and a clean code. Up to you here.

Reply:
It's easy to check without guessing. There is a difference in the generated code with and without likely with armv7 and x86_64 archs: https://godbolt.org/z/s5Tfs1jWx

Reply:
Google actively uses these attributes in its code for a reason.